nf_core/pipelines/create/utils.py

```diff
@@ -14,6 +16,22 @@
 from textual.widget import Widget
 from textual.widgets import Button, Input, Markdown, RichLog, Static, Switch
 
+# Use ContextVar to define a context on the model initialization
+_init_context_var: ContextVar = ContextVar("_init_context_var", default={})
+
+
+@contextmanager
+def init_context(value: Dict[str, Any]) -> Iterator[None]:
+    token = _init_context_var.set(value)
+    try:
+        yield
+    finally:
+        _init_context_var.reset(token)
+
+
+# Define a global variable to store the pipeline type
+PIPELINE_TYPE_GLOBAL: Union[str, None] = None
```

Comment:
How about making this an enum?

Reply:
Your comment made me think that it's actually better to use a boolean, because we only have nf-core or not nf-core, so there's no need to do complicated things :)